query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -315,6 +322,82 @@ private Filter rebuildFilterOmittingDefaults(
     return builder.clearChildFilter().addAllChildFilter(childFilters).build();
   }
 
+  private QueryRequest rebuildRequest(
+      QueryRequest original,
+      List<Expression> selections,
+      List<Expression> aggregations,
+      Filter filter,
+      List<Expression> groupBys,
+      List<OrderByExpression> orderBys) {
+
+    Set<AttributeExpression> attributeExpressionSet =
+        Stream.concat(
+                findAttributeExpressionFromOrderBy(orderBys).stream(),
+                findAttributeExpressionFromFilter(filter).stream())
+            .collect(Collectors.toSet());
+
+    return rebuildRequestOmittingDefaults(
+        original,
+        selections,
+        aggregations,
+        updateFilterForAttributeExpression(attributeExpressionSet, filter),
+        groupBys,
+        orderBys);
+  }
+
+  private List<AttributeExpression> findAttributeExpressionFromOrderBy(
+      List<OrderByExpression> orderByExpressionList) {
+    List<AttributeExpression> attributeExpressionList = new ArrayList<>();
+    for (OrderByExpression orderByExpression : orderByExpressionList) {
+      if (orderByExpression.getExpression().getValueCase() == ValueCase.ATTRIBUTE_EXPRESSION
+          && orderByExpression.getExpression().getAttributeExpression().hasSubpath()) {
+        attributeExpressionList.add(orderByExpression.getExpression().getAttributeExpression());
+      }
+    }
+    return attributeExpressionList;
+  }
+
+  private List<AttributeExpression> findAttributeExpressionFromFilter(Filter filter) {
+    List<AttributeExpression> attributeExpressionList = new ArrayList<>();
+    findAttributeExpressionFromFilter(filter, attributeExpressionList);
+    return attributeExpressionList;
+  }
+
+  private void findAttributeExpressionFromFilter(
+      Filter filter, List<AttributeExpression> attributeExpressionList) {
+
+    for (Filter childFilter : filter.getChildFilterList()) {
+      findAttributeExpressionFromFilter(childFilter, attributeExpressionList);
+    }
+
+    if (filter.getLhs().getValueCase() == ValueCase.ATTRIBUTE_EXPRESSION
+        && filter.getLhs().getAttributeExpression().hasSubpath()) {
+      attributeExpressionList.add(filter.getLhs().getAttributeExpression());
+    }
+  }
+
+  private Filter updateFilterForAttributeExpression(
+      Set<AttributeExpression> attributeExpressionSet, Filter filter) {
+
+    if (attributeExpressionSet.isEmpty()) {
+      return filter;
+    }
+
+    List<Filter> childFilterList = new ArrayList<>();
+    childFilterList.add(filter);
+
+    attributeExpressionSet.forEach(
+        attributeExpression ->
+            childFilterList.add(
+                Filter.newBuilder()
+                    .setOperator(Operator.EQ)
```

Comment:
We're still talking our query language here, so we shouldn't be doing a `__keys = value` (not only is that pinot specific but also I don't believe would translate correctly because won't show up in any of our query handler definitions). Instead, I'd expect a `<attributeId> CONTAINS_KEY <subpath>` style filter.

Reply:
resolved